### ISSUE TITLE
Machines with Webroot seem to be detecting nsis_tauri_utils.dll as a virus as of very recently

### ISSUE THREAD

**Issue by ttrushin** [amrbashir/NSIS-SemverCompare#37]
We have had a Tauri-based application in production with a few hundred users for a while, but today started to get reports of Webroot picking up the nsis_tauri_utils.dll as malware. It is preventing installation of our app. 

Here is a VirusTotal scan of the file reported to use by one of our users: https://www.virustotal.com/gui/file/c0f2b928bc4c81cc5ca30a8932a6dc8cd617dd016679c057e23355fe732b2333?nocache=1

I was informed that the issue is fixed via allowlisting in Webroot, but obviously that is not ideal. 

Just wanted to add this issue here in case anyone has any thoughts, feedback, or has experienced the same thing. 

See screenshot of error during install:
![Screenshot 2024-07-08 at 16 54 46](https://github.com/tauri-apps/nsis-tauri-utils/assets/10791881/ec4b32cc-5460-4ce3-bd27-c11a7339bd64)

**Comment by ttrushin** [comment]
Looks like this is only happening for [nsis_tauri_utils v0.4.0](https://github.com/tauri-apps/nsis-tauri-utils/releases/tag/nsis_tauri_utils-v0.4.0)

v0.3.0 did not have the issue (just checked with Webroot via their file submission tool): https://snup.webrootcloudav.com/SkyStoreFileUploader/upload.aspx#

v0.4.1 does not appear to have been picked up them yet one way or the other.

Is there any way to force Tauri to use v0.3.0, or to get v0.4.1 out so that the build process starts using it?

**Comment by FabianLars** [comment]
0.4.1 was just a small change so if they check the used win32 this shouldn't make a difference (tauri@git already updated though, will be available in the next release).

0.4 added code that uses CreateProcessW which i assume is the most problematic change there. It fixed a pretty important issue though so idk what we could do here :/
Maybe we could compile 2 versions of the dll and bundle the correct one depending on the installMode. That means the fix wouldn't be included if users manually run the perUser installer as admin but maybe that's a tradeoff worth considering...

**Comment by ttrushin** [comment]
So it sounds like the fix was made for perUser installs but is not really needed when doing perMachine installs because that will have elevated permissions anyway? 

Anything we can possibly do to not have whatever is triggering anti-virus works for us, even if it means switching to perMachine installs. 

Another question though: if we were to customize Tauri's default NSI file and remove the logic surrounding finding and killing running processes, would the DLL still get auto-included during the bundling process?

**Comment by FabianLars** [comment]
> So it sounds like the fix was made for perUser installs but is not really needed when doing perMachine installs because that will have elevated permissions anyway?

No the fix was for "perMachine" or "both" installations, perUser.

> Another question though: if we were to customize Tauri's default NSI file and remove the logic surrounding finding and killing running processes, would the DLL still get auto-included during the bundling process?

As far as i can see it'd require cli/bundler changes to not include the dll at all.

**Comment by FabianLars** [comment]
i guess another workaround for now is to downgrade the cli (if possible) to 2.0.0-beta.20 or 1.5.14

**Comment by ttrushin** [comment]
Darn I wish we could downgrade to 1.5.14. Sadly we use signCommand so we get:

Error `tauri.conf.json` error on `tauri > bundle > windows`: Additional properties are not allowed ('signCommand' was unexpected)

**Comment by ttrushin** [comment]
We are exclusively perUser. I really wish there was a way to not have to use the new DLL with the fix that technically doesn't impact our installs... 

Thanks for the back and forth with me on this, by the way. Would love to figure out a solve, but I understand that you've probably got a lot going on!

**Comment by ttrushin** [comment]
Is there a way to allow Tauri to use a specific DLL, or to customize the link (and hash, which I see the bundler uses) where it pulls the DLL from? 

I'm just trying to think of options

**Comment by FabianLars** [comment]
Let's ping @amrbashir and maybe @Legend-Master for their input. (if it's urgent i guess you'll have to fork the repo and use the cli from git for now)

**Comment by Legend-Master** [comment]
Honestly I don't know which part triggered it, if `CreateProcess` is the problem then how is NSIS's `nsExec` not a problem which uses `ShellExecute` under the hood

**Comment by FabianLars** [comment]
I think it's the combination of writing a dll that contains shellexecute to disk at runtime, similar to how many antivir softwares have a problem with rust's build scripts.

**Comment by ttrushin** [comment]
I'm surprised more people aren't chiming in saying they have the same issue. Every time that we install our software on a new user's machine that has Webroot (not common, but certainly not uncommon on corporate machines), installation fails due to this DLL. 

We've been allow-listing the DLL by its hash and reattempting installation with success, but it is definitely quite frustrating.

**Comment by FabianLars** [comment]
Yeah, i've not received any complaints in other communication channels either.

**Comment by ajladybug** [comment]
FWIW, I am also unable to install due to the same error and Webroot.

**Comment by ttrushin** [comment]
We have been manually allow-listing the DLL and reattempts are successful, but it is very annoying, and occasionally very difficult when working with enterprise orgs (where getting IT on the line at the time of install can be time consuming).

**Comment by amrbashir** [comment]
I am replacing `nsis_tauri_utils.dll` usage with `nsis_semvercompare.dll` and `nsProcess.dll` which both shouldn't have any problem with anti viruses, could you give it a try https://github.com/tauri-apps/tauri/pull/11041?

**Comment by amrbashir** [comment]
I have re-checked nsis-tauri-utils@0.4.1 and it passes all anti viruses https://www.virustotal.com/gui/url/e814a313d4ba77fd51c3d1408c3ba49329a52656a767df50bdcbb867bcf837f6?nocache=1 and passes Webroot as well.

cc @ttrushin

**Comment by setoelkahfi** [comment]
Let me know if this isn't related.

I have published 10+ apps on the Microsoft Store with pretty much the same Tauri config but all of sudden got this rejection review:

<img width="1510" height="302" alt="Image" src="https://github.com/user-attachments/assets/1624b661-edbe-4a65-9777-85798c401159" />

**Comment by Elendiar** [comment]
After updating to `tauri-cli 🦀: 2.10.0` can't build with 
```sh
Warn NSIS directory is missing some files. Recreating it.
        Info Verifying NSIS package
 Downloading https://github.com/tauri-apps/binary-releases/releases/download/nsis-3.11/nsis-3.11.zip
        Info validating hash
        Info extracting NSIS
 Downloading https://github.com/tauri-apps/nsis-tauri-utils/releases/download/nsis_tauri_utils-v0.5.3/nsis_tauri_utils.dll
        Info validating hash
        Info Target: x64
 Running makensis to produce ...
Error failed to bundle project `The system cannot find the file specified. (os error 2)`
```
`WithSecure Elements Agent` remove harmful file 
`C:\Users\***\AppData\Local\tauri\NSIS\Plugins\x86-unicode\additional\nsis_tauri_utils.dll` with reason `Trojan.TR/Hijacker.Gen`

**Comment by Legend-Master** [comment]
Yeah... For tauri-cli 2.10, we have changed our way to run the app non-elevating in #53 to use a token impersonation approach that's also used by the Chromium installer, since this approach can be used the other way around, some antivirus software might detect it as a virus

I have tried to upload it to VirusTotal and it seems to pass all the tests though: https://www.virustotal.com/gui/url/81ec5d001990d176e92996c0664e7943d024fe9f8c06cbf24874ea11a5c92cc3/detection

**Comment by len360** [comment]
I am still experiencing the issue despite having all dependencies up to date. This is problematic for deploying the application in an enterprise environment. Has anyone found a workaround?
